L46-python-compression-api.md

```diff
@@ -0,0 +1,196 @@
+the following constants.
+
+```python
+NoCompression = _cygrpc.GRPC_COMPRESS_NONE
+Deflate = _cygrpc.GRPC_COMPRESS_DEFLATE
+Gzip = _cygrpc.GRPC_COMPRESS_GZIP
+StreamGzip = _cygrpc.GRPC_COMPRESS_STREAM_GZIP
+```
+
+These contants mirror those in gRPC core.
+
+In addition, I propose adding a `compression` keyword argument to the
+`__call__`, `with_call`, and `future` methods of the `XXXMultiCallable` family
+of interfaces. This argument will set the compression method to use for an
+individual call.
+
+```python
+class UnaryUnaryMultiCallable(six.with_metaclass(abc.ABCMeta)):
+
+    @abc.abstractmethod
+    def __call__(self,
+                 request,
+                 timeout=None,
+                 metadata=None,
+                 credentials=None,
+                 wait_for_ready=None,
+                 compression=None):
+        ...
+
+    @abc.abstractmethod
+    def with_call(self,
+                  request,
+                  timeout=None,
+                  metadata=None,
+                  credentials=None,
+                  wait_for_ready=None,
+                  compression=None):
+        ...
+
+    @abc.abstractmethod
+    def future(self,
+               request,
+               timeout=None,
+               metadata=None,
+               credentials=None,
+               wait_for_ready=None,
+               compression=None):
+        ...
+
+
+class UnaryStreamMultiCallable(six.with_metaclass(abc.ABCMeta)):
+
+    @abc.abstractmethod
+    def __call__(self,
+                 request,
+                 timeout=None,
+                 metadata=None,
+                 credentials=None,
+                 wait_for_ready=None,
+                 compression=None):
+        ...
+
+
+class StreamUnaryMultiCallable(six.with_metaclass(abc.ABCMeta)):
+
+    @abc.abstractmethod
+    def __call__(self,
+                 request_iterator,
+                 timeout=None,
+                 metadata=None,
+                 credentials=None,
+                 wait_for_ready=None,
+                 compression=None):
+        ...
+
+    @abc.abstractmethod
+    def with_call(self,
+                  request_iterator,
+                  timeout=None,
+                  metadata=None,
+                  credentials=None,
+                  wait_for_ready=None,
+                  compression=None):
+        ...
+
+    @abc.abstractmethod
+    def future(self,
+               request_iterator,
+               timeout=None,
+               metadata=None,
+               credentials=None,
+               wait_for_ready=None,
+               compression=None):
+        ...
+
+
+class StreamStreamMultiCallable(six.with_metaclass(abc.ABCMeta)):
+
+    @abc.abstractmethod
+    def __call__(self,
+                 request_iterator,
+                 timeout=None,
+                 metadata=None,
+                 credentials=None,
+                 wait_for_ready=None,
+                 compression=None):
+        ...
+```
```

Comment:
The interceptor API interface have to change as well. Don't forget the [ClientCallDetails](https://grpc.io/grpc/python/grpc.html#grpc.ClientCallDetails).